### PR TITLE
VP-7061: Move auth requirement to Data project

### PR DESCRIPTION
**Pull request by tatarincev**
### Description:
- Move auth requirement to Data project
- Add sample how to extend existing auth policy with new custom policies

### Jira: https://virtocommerce.atlassian.net/browse/VP-7061

### CODE REVIEW

**Review by kostyrin — COMMENTED** [review]
`HandleRequirementAsync` looks like generic

File: samples/VirtoCommerce.OrdersModule2.Web/Authorization/CustomOrderAuthorizationHandler.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+using VirtoCommerce.OrdersModule.Core.Model.Search;
+using VirtoCommerce.OrdersModule.Data.Authorization;
+using VirtoCommerce.Platform.Core.Security;
+using VirtoCommerce.Platform.Security.Authorization;
+
+namespace VirtoCommerce.OrdersModule2.Web.Authorization
+{
+    public sealed class CustomOrderAuthorizationHandler : PermissionAuthorizationHandlerBase<OrderAuthorizationRequirement>
+    {
+
```
Comment (ksavosteev) [inline samples/VirtoCommerce.OrdersModule2.Web/Authorization/CustomOrderAuthorizationHandler.cs:16]: remove empty line

File: samples/VirtoCommerce.OrdersModule2.Web/Module.cs [hunk 2]
```diff
@@ -25,13 +28,24 @@ public void Initialize(IServiceCollection serviceCollection)
                 options.UseSqlServer(configuration.GetConnectionString(ModuleInfo.Id) ?? configuration.GetConnectionString("VirtoCommerce.Orders") ?? configuration.GetConnectionString("VirtoCommerce"));
             });
             serviceCollection.AddTransient<IOrderRepository, OrderRepository2>();
+
+
```
Comment (ksavosteev) [inline samples/VirtoCommerce.OrdersModule2.Web/Module.cs:32]: remove empty line

File: samples/VirtoCommerce.OrdersModule2.Web/Module.cs [hunk 3]
```diff
@@ -25,13 +28,24 @@ public void Initialize(IServiceCollection serviceCollection)
...
             var settingsRegistrar = appBuilder.ApplicationServices.GetRequiredService<ISettingsRegistrar>();
             settingsRegistrar.RegisterSettings(ModuleConstants.Settings.General.AllSettings, ModuleInfo.Id);
 
+            AbstractTypeFactory<PermissionScope>.RegisterType<OrderSelectedStatusScope>();
+            var permissionsProvider = appBuilder.ApplicationServices.GetRequiredService<IPermissionsRegistrar>();
+            permissionsProvider.WithAvailabeScopesForPermissions(new[] {
+                                                                        OrdersModule.Core.ModuleConstants.Security.Permissions.Read,
+                                                                        }, new OrderSelectedStatusScope());
+
+
```
Comment (ksavosteev) [inline samples/VirtoCommerce.OrdersModule2.Web/Module.cs:48]: remove empty line

File: samples/VirtoCommerce.OrdersModule2.Web/Scripts/order2.js [hunk 4]
```diff
@@ -6,53 +6,74 @@ if (AppDependencies != undefined) {
...
-        knownOperations.registerOperation(invoiceOperation);
-
-        $http.get('Modules/$(virtoCommerce.orders2)/Scripts/tree-template.html').then(function (response) {
-            // compile the response, which will put stuff into the cache
-            $compile(response.data);
-        });
-    }]);
+                };
+                scopeResolver.register(orderStatusesScope);
+
```
Comment (ksavosteev) [inline samples/VirtoCommerce.OrdersModule2.Web/Scripts/order2.js:78]: remove empty line

File: samples/VirtoCommerce.OrdersModule2.Web/Module.cs [hunk 5]
```diff
@@ -25,13 +28,24 @@ public void Initialize(IServiceCollection serviceCollection)
...
         public void PostInitialize(IApplicationBuilder appBuilder)
         {
             var settingsRegistrar = appBuilder.ApplicationServices.GetRequiredService<ISettingsRegistrar>();
             settingsRegistrar.RegisterSettings(ModuleConstants.Settings.General.AllSettings, ModuleInfo.Id);
 
+            AbstractTypeFactory<PermissionScope>.RegisterType<OrderSelectedStatusScope>();
+            var permissionsProvider = appBuilder.ApplicationServices.GetRequiredService<IPermissionsRegistrar>();
+            permissionsProvider.WithAvailabeScopesForPermissions(new[] {
+                                                                        OrdersModule.Core.ModuleConstants.Security.Permissions.Read,
+                                                                        }, new OrderSelectedStatusScope());
```
Comment (eMazeika) [inline samples/VirtoCommerce.OrdersModule2.Web/Module.cs:46]: need description what this sample demonstrates. In readme/module.cs/docs?

Comment (Vectorfield4) [reply samples/VirtoCommerce.OrdersModule2.Web/Module.cs]: Description added.

File: samples/VirtoCommerce.OrdersModule2.Web/Module.cs (hunk 3 above)
Comment (Vectorfield4) [reply samples/VirtoCommerce.OrdersModule2.Web/Module.cs:54]: Line removed.

File: samples/VirtoCommerce.OrdersModule2.Web/Module.cs (hunk 2 above)
Comment (Vectorfield4) [reply samples/VirtoCommerce.OrdersModule2.Web/Module.cs:36]: Line removed.

File: samples/VirtoCommerce.OrdersModule2.Web/Authorization/CustomOrderAuthorizationHandler.cs (hunk 1 above)
Comment (Vectorfield4) [reply samples/VirtoCommerce.OrdersModule2.Web/Authorization/CustomOrderAuthorizationHandler.cs]: Line removed.

File: samples/VirtoCommerce.OrdersModule2.Web/Scripts/order2.js (hunk 4 above)
Comment (Vectorfield4) [reply samples/VirtoCommerce.OrdersModule2.Web/Scripts/order2.js]: Removed.

**Review by Vectorfield4 — APPROVED** [review]

**Review by kostyrin — APPROVED** [review]

**Review by AliveMen — APPROVED** [review]

**Review by ksavosteev — APPROVED** [review]

**Review by eMazeika — APPROVED** [review]

**Pull request merged**